### PR TITLE
Expose jetty request-header-size and response-header-size as command …

### LINKED ISSUE #9386: Azure Web App AD Authentication causes - Request Header Fields Too Large
I'm using Metabase Azure Web App with the metabase/metabase docker image. We have enabled prelogin with Azure Active directory which attaches a cookie to the request. This often results in:

Bad Message 431
reason: Request Header Fields Too Large

I'm not 100% sure, but I presume this is configured on the metabase side to increase the header size.

image

- Your browser and the version: Chrome
- Your operating system: macOS (mine)/system (whatever Azure use)
- Your databases: MySQL
- Metabase version: v0.31.2/latest
- Metabase hosting environment: Azure Web App container - Docker Hub (metabase/metabase)
- Metabase internal database: H2 (default)

### PR DESCRIPTION
**Pull request by juahan**
Fixes #9386

Fixed issue is an issue when in some cases Jetty complains about too large headers. Specifically this happens when the Metabase docker used in combination with Azure and Active Directory authentication. With this combination sometimes the authentication headers grow too long and Jetty will give "Bad Message 431, reason: Request Header Fields Too Large" error. This is a protection for the server so that it cannot be bogged down with too large headers.

The proposed fix is simple. It exposes the Jetty request-header-size and response-header-size parameters to the docker configuration and sets their default as 8192.

### CODE REVIEW

File: src/metabase/server.clj [hunk 1]
```diff
@@ -23,12 +23,14 @@
...
-            :min-threads   (config/config-int :mb-jetty-minthreads)
-            :max-queued    (config/config-int :mb-jetty-maxqueued)
-            :max-idle-time (config/config-int :mb-jetty-maxidletime)})
+           {:port                 (config/config-int :mb-jetty-port)
+            :host                 (config/config-str :mb-jetty-host)
+            :max-threads          (config/config-int :mb-jetty-maxthreads)
+            :min-threads          (config/config-int :mb-jetty-minthreads)
+            :max-queued           (config/config-int :mb-jetty-maxqueued)
+            :request-header-size  (config/config-int :mb-jetty-maxrequestheader)
+            :response-header-size (config/config-int :mb-jetty-maxresponseheader)
```
Comment (walterl) [inline src/metabase/server.clj]: Could we rather keep with the naming scheme of the other variables? That'll strengthen the Metabase options' references to the specific Jetty options.

 ```suggestion
             :request-header-size  (config/config-int :mb-jetty-requestheadersize)
             :response-header-size (config/config-int :mb-jetty-responseheadersize)
```

**Review by walterl — CHANGES_REQUESTED** [review]
Thank you so much for this PR and all your work on #9386! :tada:

Besides the one comment I added, I see there's [a test failing](https://circleci.com/gh/metabase/metabase/229961) due to your changes. You probably just need to add those new options of yours in [this test](https://github.com/juahan/metabase/blob/header-size/test/metabase/server_test.clj) with the value `10` for each.

File: src/metabase/server.clj (hunk 1 above)
Comment (camsaul) [reply src/metabase/server.clj]: Any reason to not follow our convention of not combining separate words into one mega-word?

This PR adds things like `MB_JETTY_REQUESTHEADERSIZE`, I personally think `MB_JETTY_REQUEST_HEADER_SIZE` is a bit more readable. Even better would be something like `MB_JETTY_REQUEST_HEADER_SIZE_BYTES` that makes it clear at a glance what unit to use

Comment (camsaul) [reply src/metabase/server.clj]: @walterl I think in this case `MB_JETTY_MAX_REQUEST_HEADER_SIZE_BYTES` would be clearest. `:request-header-size` isn't a name used internally in Jetty. It's the name Ring uses for some Jetty setting somewhere. So just because Ring names the Jetty option `:request-header-size` doesn't mean we have to follow suit and use an ambiguous name for a Jetty option.

I'd rather use clear and unambiguous stuff even if it doesn't line up 100% with the param names used by Ring

Comment (walterl) [reply src/metabase/server.clj]: I'll similarly update the options around these new ones, once this PR is merged. :+1:

Comment (juahan) [reply src/metabase/server.clj]: Sorry for the delay. I agree with the discussed changes, made chagnges according to the discussions and have now committed them. 

This is my first push request ever, so if I'm completely honest, not sure what do I have to do now.

**Review by walterl — CHANGES_REQUESTED** [review]
Hi again @juahan. Your last change made the test pass! :tada: 

I have one more tiny code style related request, but that's it. Once the indicated underscores are replaced with dashes (as we do in Clojure land), this PR is ready to merge :sunglasses:.

File: src/metabase/server.clj [hunk 2]
```diff
@@ -23,12 +23,14 @@
...
-            :min-threads   (config/config-int :mb-jetty-minthreads)
-            :max-queued    (config/config-int :mb-jetty-maxqueued)
-            :max-idle-time (config/config-int :mb-jetty-maxidletime)})
+           {:port                 (config/config-int :mb-jetty-port)
+            :host                 (config/config-str :mb-jetty-host)
+            :max-threads          (config/config-int :mb-jetty-maxthreads)
+            :min-threads          (config/config-int :mb-jetty-minthreads)
+            :max-queued           (config/config-int :mb-jetty-maxqueued)
+            :request-header-size  (config/config-int :mb-jetty-max_request_header_size_bytes)
+            :response-header-size (config/config-int :mb-jetty-max_response_header_size_bytes)
```
[the remark this answers is not in the document]
Comment (juahan) [reply src/metabase/server.clj]: Underscores should now be changed to dashes.

File: src/metabase/config.clj [hunk 3]
```diff
@@ -23,6 +23,8 @@
    ;; Jetty Settings. Full list of options is available here: https://github.com/ring-clojure/ring/blob/master/ring-jetty-adapter/src/ring/adapter/jetty.clj
    :mb-jetty-port          "3000"
    :mb-jetty-join          "true"
+   :mb-jetty-max_request_header_size_bytes  "8192"
+   :mb-jetty-max_response_header_size_bytes "8192"
```
[the remark this answers is not in the document]
Comment (juahan) [reply src/metabase/config.clj]: Should be changed, I don't really understand why I missed that when all other variables in there are with dashes :laughing:

File: src/metabase/config.clj [hunk 4]
```diff
@@ -23,6 +23,8 @@
    ;; Jetty Settings. Full list of options is available here: https://github.com/ring-clojure/ring/blob/master/ring-jetty-adapter/src/ring/adapter/jetty.clj
    :mb-jetty-port          "3000"
    :mb-jetty-join          "true"
+   :mb-jetty-max-request-header-size-bytes  "8192"
+   :mb-jetty-max-response-header-size-bytes "8192"
```
Comment (walterl) [inline src/metabase/config.clj]: I did a final live test and while running this code I realized that these defaults should probably be removed: if the defaults are specified here they will always override Ring's defaults. In other words, if the Ring defaults should change in the future, these defaults will represent an unexpected deviation. Instead it's better to remove these defaults, so that the values are only set when the relevant `MB_JETTY_*` environment variable is set.

TL;DR: Just remove these lines :stuck_out_tongue:. The rest is great.

```suggestion
```

**Review by walterl — CHANGES_REQUESTED** [review]
I overlooked the effect of specifying the defaults before, and really think we should try and limit the side effects of this change as much as possible.

**Comment by flamber** [comment]
Closed in favor of #12683
